v3/hpc_symmetric3x3.hpp

```diff
@@ -278,12 +281,43 @@ trace(symmetric3x3<T> const x) noexcept {
   return x(S_XX) + x(S_YY) + x(S_ZZ);
 }
 
+template <class T>
+HPC_ALWAYS_INLINE HPC_HOST_DEVICE constexpr symmetric3x3<T>
+iso(symmetric3x3<T> const x) noexcept {
+  return ((1.0 / 3.0) * trace(x)) * symmetric3x3<T>::identity();
+}
+
 template <class T>
```

Comment:
@ibaned I really don't like the construct `x - ((1.0 - 3.0) * trace(x)` because it violates the cardinal rule that tensor operations must be consistent with respect to their indices.  What does subtracting a scalar from a matrix even mean?  In numpy, the scalar is subtracted from every value, here we subtract it from just the diagonals.  I'd much prefer the construct in this PR.

Reply:
I think there is more benefit than harm to saying that math operations between a tensor and a scalar behaves as though the scalar is multiplied first by the identity tensor. Mathematicians do this (sigma minus p) and there are cases that can avoid additional memory ops `sigmas[point] += p`.

Reply:
On the topic of reducing operations - I’m in 100% agreement.  But, dimensional analysis is just as important as unit analysis.  Just as it does not make sense to add a meter to a second, it makes no sense to add a zeroth order tensor (scalar) to a nth order tensor (n!=0).  I don’t think an overload should exist for this case.  For the computation of the deviator, I’d rather see the trace subtracted explicitly from the diagonals than dispatch it to an overload.  `sigma_ij - p` is ill posed, and I can’t imagine a reviewer letting it sneak through peer review - I don’t believe the code should let it sneak through either

Reply:
On the topic of reducing operations - I’m in 100% agreement.  But, dimensional analysis is just as important as unit analysis.  Just as it does not make sense to add a meter to a second, it makes no sense to add a zeroth order tensor (scalar) to a nth order tensor (n!=0).  I don’t think an overload should exist for this case.  For the computation of the deviator, I’d rather see the trace subtracted explicitly from the diagonals than dispatch it to an overload.  `sigma_ij - p` is ill posed, and I can’t imagine a reviewer letting it sneak through peer review - I don’t believe the code should let it sneak through either

Reply:
Okay, the last commit should have less complexity